runtime/lua/vim/snippet.lua

```diff
@@ -222,6 +224,26 @@ end
 --- @field private _session? vim.snippet.Session
 local M = { session = nil }
 
+--- Displays the choices for the given tabstop as completion items.
+---
+--- @param tabstop vim.snippet.Tabstop
+local function display_choices(tabstop)
+  assert(tabstop.choices, 'Tabstop has no choices')
+
+  local start_col = tabstop:get_range()[2] + 1
+  local matches = vim.iter.map(function(choice)
+    return {
+      word = choice,
+      abbr = choice,
+      menu = '[Snippet]',
+    }
+  end, tabstop.choices)
+
+  vim.defer_fn(function()
+    vim.fn.complete(start_col, matches)
+  end, 100)
```

Comment:
This use of `vim.defer_fn` follows `nvim-snippy`, which does this because ["some autocomplete plugins may close the pum menu to present their own completion items"](https://github.com/dcampos/nvim-snippy/issues/16#issuecomment-953016668). I'm not sure if this is still an issue or something that we should consider in this implementation though...